### ISSUE TITLE
Idle Notifier: False idle notifications on low alert threshold for some animations

### ISSUE THREAD

**Issue by tylerwgrass** [runelite/runelite#18010]
### Type

Incorrect behavior

### Operating System

None

### Bug description

Some actions do not consistently perform their associated animation when occurring continuously.  
If the idle notifier threshold is low enough, it can falsely send a notification that the character is idle.

For example, while blowing an inventory of glass, sometimes the animation appears as idle for ~2-3 ticks despite blowing the glass.  This means if the idle threshold is set to 1 second, it will send an idle notification (see video).

Some considered fixes:
1. Somehow determine an action is being performed without relying on the animation (preferred, but not sure if possible)
2. Set a minimum required notification wait period (not recommended)
3. Add a small time buffer to account for these sort of animations (not recommended)

Appreciate recommendations, thanks!

### Screenshots or videos

Included animation ids in the chat, 884 is blow glass, -1 is idle

https://github.com/runelite/runelite/assets/22583776/76b333dd-c0c9-4eee-bac5-075788a8a913

### RuneLite version

Launcher version: Version 0.34.2 (0.34.2)
RuneLite version: 1.10.34-SNAPSHOT

**Comment by Nightfirecat** [comment]
That's not really a false positive, you should set your delay higher.

**Comment by tylerwgrass** [comment]
Yes, I have done that.

However, I would still argue that this is not intended behavior. I believe the intent of the plugin is to notify the player when they are no longer performing an action and may require input from them.  In this case, it sends a notification despite the fact the user is not actually idle.

**Comment by tylerwgrass** [comment]
Assuming there's no way to determine actions like these are occurring other than via the animation, enforcing a minimum "Idle Notification Delay" seems sufficient

**Comment by Felanbird** [comment]
There's already a default `Idle Notifier Delay` value that's more than efficient to prevent this, you have to change it yourself to end up in this scenario, so I don't see why we would need to enforce something on users

**Comment by tylerwgrass** [comment]
But a default is not a requirement. While I would understand in cases where a user sets a setting to "ping me ever 1ms" the answer may be "don't do that", I don't think that is a valid approach in all cases. As a user I would not expect that if I set my alert threshold to be 3 ticks that this would be too low, as it is valid for most use cases.

It's also not obvious why you receive these alerts with a low-ish threshold unless you view the source code and see it relies on the characters current animation as opposed to the action being performed.


Given that the initial recommendation provided above was "increase your delay" it doesn't seem illogical to me to have a minimum to prevent others who may potentially open a separate issue just to get the same answer.

**Comment by Felanbird** [comment]
As a note, it was mentioned in the discord during this conversation that glass blowing is one of the slowest in the game, meaning the issue you're having is an extreme example

**Comment by Felanbird** [comment]
Also, the hub plugin [action progress](https://runelite.net/plugin-hub/show/action-progress) should do exactly what you want without the erroneous idle notification problems, for this slow animation

**Comment by tylerwgrass** [comment]
Thanks for the input. I just want to clarify a few things. (As I just viewed some of the discord conversation as well).

1) My intent with opening this was to simply report something that to me seemed unintended. The tone did not mean to come off as "this is an issue I cannot overcome and needs to be fixed!!" (I apologize if that came off that way, given your messages in the discord)

2) I think action progress bars is an alternative but doesn't necessarily address this issue. 

Consider the use case where you want to AFK glass glowing with client on your second monitor and screen flashes enabled.  With this setup in the past, I've encountered upwards of ~3 screen flash notifications within an inventory of blowing glass.  The intent isn't to watch the screen, wait for the action to complete by watch the progress, then perform another action.

I understand this is an edge case and did not intend for this to require as deep conversation as it has already gone. I would participate in the discord conversation as well but I understand not having permissions for that ☺️.
